sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -112,7 +112,12 @@ class SimpleFunctionRegistry extends FunctionRegistry {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    try {
+      func(children)
```

Comment:
This is a scala function. It might be harder to do it. 

If using the case match, it might be looks like?
```
    func match {
      case f: Function0[_] if children.size != 0 => ...
      case f: Function1[_, _] if children.size != 1 => ...
      ...
```